Проектная работа 1

aza1rat opened the pull request.

aza1rat committed:
First commit

aza1rat committed:
Added speed parsing

aza1rat committed:
Changed README

ArturNurtdinov commented on src/main/java/Automobile.java:
```diff
@@ -0,0 +1,20 @@
+    public int getDistance()
+    {
+        return distance;
+    }
```
Так как поля помечены `final`, то геттеры можно убрать, а поля пометить как публичные, чтобы получать доступ при чтении поля к нему напрямую, а не через геттер. Это безопасно, так как поле помечено как `final`

ArturNurtdinov commented on src/main/java/Main.java:
```diff
@@ -1,6 +1,49 @@
 
 public class Main {
+    private final static int automobileCount = 3;
+    private final static int maxSpeed = 250;
```
В Java и Kotlin константы принято называть в соответствии со SCREAMING_SNAKE_CASE, т.е. здесь лучше переименовать в:
```suggestion
    private final static int AUTOMOBILE_COUNT= 3;
    private final static int MAX_SPEED = 250;
```

ArturNurtdinov commented on src/main/java/Race.java:
```diff
@@ -0,0 +1,25 @@
+public class Race {
+    private static Automobile winner;
```
Не стоит писать все классы и функции исключительно статическими, так как это ограничивает гибкость кода и делает его трудным для тестирования и масштабирования. Лучше использовать объектно-ориентированный подход - разбить логику на классы и создавать экземпляры этих классов, чтобы каждая часть отвечала за свою зону ответственности.

ArturNurtdinov commented on src/main/java/Race.java:
```diff
@@ -0,0 +1,25 @@
+        return duration;
+    }
+
+    public static void CheckWinner(Automobile automobile) {
```
В Java и Kotlin методы классов принято именовать с маленькой буквы

aza1rat committed:
Update src/main/java/Main.java

Co-authored-by: Artur <49075521+ArturNurtdinov@users.noreply.github.com>

aza1rat committed:
After review:
- Changed naming of constants and methods
- Class Race now non-static
- Removed getters (there are public final fields)
